src/librustc_trans/glue.rs

```diff
@@ -338,9 +338,9 @@ pub fn size_and_align_of_dst<'blk, 'tcx>(bcx: &BlockAndBuilder<'blk, 'tcx>,
             // Returns size in bytes of all fields except the last one
             // (we will be recursing on the last one).
             fn local_prefix_bytes(variant: &ty::layout::Struct) -> u64 {
-                let fields = variant.offset_after_field.len();
+                let fields = variant.offsets.len();
                 if fields > 1 {
-                    variant.offset_after_field[fields - 2].bytes()
+                    variant.offsets[fields - 1].bytes()
                 } else {
                     0
                 }
```

Comment:
This whole thing can be `variant.offsets.last().map_or(0, |o| o.bytes())` - and `local_prefix_bytes` doesn't even need to exist as a separate function.